src/components/Common/ImageUpload/useUploadImages.jsx

```diff
@@ -0,0 +1,199 @@
+import axios from "axios";
+import { useState } from "react";
+import { toast } from "react-toastify";
+import { getApiUrl } from "src/features/apiUrl";
+
+const useUploadImages = ({ maxImageSizeByte, pageName, initImages = [] }) => {
+    const [images, setImages] = useState([]);
+    const [imagesName, setImagesName] = useState(initImages || []);
+
+    const handleValidate = (image, maxSizeByte) => {
+        if (!image) return;
+        if (!image.type.startsWith("image/") && !["image/png", "image/jpeg", "image/jpg"].includes(image.type)) {
+            toast.error("Invalid image type. Only png, jpeg and jpg are allowed.");
+            return;
+        }
+        if (image.size > maxSizeByte) {
+            toast.error(`Image size should be less than ${maxSizeByte / 1000}KB.`);
+            return;
+        }
+        return image;
+    };
+
+    const handleUploadImage = (image) => {
+        if (!handleValidate(image, maxImageSizeByte)) return;
+        const imageName = `${pageName}-${Date.now()}.${image?.type.split("/")[1]}`;
+        return new Promise((resolve, reject) => {
+            const reader = new FileReader();
+            reader.onloadend = () => {
+                resolve({
+                    image: new File([reader.result], imageName, { type: image?.type }),
+                    imageName: imageName.split("-")[1],
+                });
+            };
+            reader.readAsArrayBuffer(image);
+        });
+    };
+
+    const handleManyUploads = async (imageFiles, maxImages = undefined) => {
+        const manyImages = [];
+        const manyImagesName = [];
+        try {
```

Comment:
![image](https://github.com/thecyberworld/TheCyberHUB/assets/76536506/85dbb77d-a12f-41fa-8947-bbc8d3044a7b)

I can't, it may be confusing because I use those names part of state